src/main/java/org/radarcns/service/DataSetService.java

```diff
@@ -280,7 +265,8 @@ public AggregatedDataPoints getDistinctData(String projectName, String subjectId
         try {
             for (AggregateDataSource source : sources) {
                 for (SourceData sourceData : source.getSourceData()) {
-                    SourceDataDTO definition = sourceCatalog.getSourceData(sourceData.getName());
+                    SourceDataDTO definition = this.sourceCatalog.getSourceData(sourceData
+                            .getName()).getSourceData();
```

Comment:
Same if changed above, this can be getSourceDataDTO. Will make it clearer

Reply:
I wouldn't name a method with DTO since it is not nice java coding convention.

Reply:
Okay then maybe the `getSourceData()` in SourceCatalog can be renamed to `getSourceDataWrapper()` ?